### PR TITLE
SM monitor shows total moles & SM gets it's own defines file.

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
Ever wondered why you failed to setup the scrungularity?

Well, wonder no more! The NT CIMS tab has been made to show total moles. The pressure got kicked however, since it's practically useless. The bar also turns red if your SM is eligible for scrung.

Moved the supermatter defines into it's own files too.

## Why It's Good For The Game
Displays a more useful metric, lets players know why their SM didnt go scrung. Define files are good.

## Changelog
:cl:
qol: Supermatter Monitor now shows total moles. It glows red if your SM is going to singulo.
code: Supermatter's defines have been moved into it's own file in the __DEFINES folder.
/:cl:

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/sm_monitor.dm [hunk 1]
```diff
@@ -130,12 +130,13 @@
...
 		data["SM_ambienttemp"] = air.temperature
-		data["SM_ambientpressure"] = air.return_pressure()
+		data["SM_bad_moles_amount"] = MOLE_PENALTY_THRESHOLD / active.gasefficency 
+		data["SM_moles"] = 0
 		//data["SM_EPR"] = round((air.total_moles / air.group_multiplier) / 23.1, 0.01)
 		var/list/gasdata = list()
 
-
 		if(air.total_moles())
+			data["SM_moles"] = air.total_moles()
```
Comment (LemonInTheDark) [inline code/modules/modular_computers/file_system/programs/sm_monitor.dm:139]: Mind yeeting that comment? It's not even applicable to our codebase, much less our current implementation of the supermatter.

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
A request for cleanup in ui_data(), although I'd also like to mention I don't like the idea of moving those sm defines over to a unique file, since they both shouldn't be used outside of the supermatter.dm file, and are on the whole the balancing numbers for the engine. it's useful to have them in the same file as the things they effect.

**Comment by stylemistake** [comment]
I don't like that you've yeeted a pressure metric, which is useful sometimes. More metrics always better.

**Comment by Acer202** [comment]
> I don't like that you've yeeted a pressure metric, which is useful sometimes. More metrics are always better.

I agree. What if you added a fancy new TGUI pressure gauge to the SM monitor instead of axing the pressure reading? The more it _feels_ like you're in the Chernobyl control room when things are going wrong, the better.

**Comment by vincentiusvin** [comment]
Alright, a few updates. I added an average section for the total moles count that starts at 0.75, since i feel like half is cutting it too short, hope you wont mind this bit?

I readded the pressure metric, but I'm not going to implement the gauge. It feels out of place and ends up cluttering.

Cleaned the sm_monitor.dm bit, but I'm not sure on how to proceed for the defines part. One of the entries in this PR requires one of the defines used by the SM, should i just substitute it for a normal integer? And isn't it the policy to try and move defines to the start of the compile list? I'm not sure i fully understand the reasoning here yet.

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -1,120 +1,3 @@
...
-//Zap constants, speeds up targeting
-
-#define BIKE (COIL + 1)
-#define COIL (ROD + 1)
-#define ROD (LIVING + 1)
-#define LIVING (MACHINERY + 1)
-#define MACHINERY (OBJECT + 1)
-#define OBJECT (LOWEST + 1)
-#define LOWEST (1)
-
```
Comment (LemonInTheDark) [inline code/modules/power/supermatter/supermatter.dm:14]: God damn it you're right, don't move this comment and these defines tho, they should both be local to this file.

Comment (vincentiusvin) [reply code/modules/power/supermatter/supermatter.dm]: Alright, moved it around. Hope this is fine now

**Pull request merged**